entgql/internal/renamedtype/ent/gql_node.go

```diff
@@ -0,0 +1,259 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Code generated by entc, DO NOT EDIT.
+
+package ent
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"entgo.io/contrib/entgql"
+	"entgo.io/contrib/entgql/internal/renamedtype/ent/clashingtext"
+	"entgo.io/ent/dialect"
+	"entgo.io/ent/dialect/sql"
+	"entgo.io/ent/dialect/sql/schema"
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/hashicorp/go-multierror"
+	"golang.org/x/sync/semaphore"
+)
+
+// Noder wraps the basic Node method.
+type Noder interface {
+	Node(context.Context) (*Node, error)
+	IsNode()
+}
+
+// IsNode implements the Node interface check for GQLGen.
+func (n *ClashingText) IsNode() {}
+
+var errNodeInvalidID = &NotFoundError{"node"}
+
+// NodeOption allows configuring the Noder execution using functional options.
+type NodeOption func(*nodeOptions)
+
+// WithNodeType sets the node Type resolver function (i.e. the table to query).
+// If was not provided, the table will be derived from the universal-id
+// configuration as described in: https://entgo.io/docs/migrate/#universal-ids.
+func WithNodeType(f func(context.Context, int) (string, error)) NodeOption {
+	return func(o *nodeOptions) {
+		o.nodeType = f
+	}
+}
+
+// WithFixedNodeType sets the Type of the node to a fixed value.
+func WithFixedNodeType(t string) NodeOption {
+	return WithNodeType(func(context.Context, int) (string, error) {
+		return t, nil
+	})
+}
+
+type nodeOptions struct {
+	nodeType func(context.Context, int) (string, error)
+}
+
+func (c *Client) newNodeOpts(opts []NodeOption) *nodeOptions {
+	nopts := &nodeOptions{}
+	for _, opt := range opts {
+		opt(nopts)
+	}
+	if nopts.nodeType == nil {
+		nopts.nodeType = func(ctx context.Context, id int) (string, error) {
+			return c.tables.nodeType(ctx, c.driver, id)
+		}
+	}
+	return nopts
+}
+
+// Noder returns a Node by its id. If the NodeType was not provided, it will
+// be derived from the id value according to the universal-id configuration.
+//
+//	c.Noder(ctx, id)
+//	c.Noder(ctx, id, ent.WithNodeType(typeResolver))
+func (c *Client) Noder(ctx context.Context, id int, opts ...NodeOption) (_ Noder, err error) {
+	defer func() {
+		if IsNotFound(err) {
+			err = multierror.Append(err, entgql.ErrNodeNotFound(id))
+		}
+	}()
+	table, err := c.newNodeOpts(opts).nodeType(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	return c.noder(ctx, table, id)
+}
+
+func (c *Client) noder(ctx context.Context, table string, id int) (Noder, error) {
+	switch table {
+	case clashingtext.Table:
+		query := c.ClashingText.Query().
+			Where(clashingtext.ID(id))
+		query, err := query.CollectFields(ctx, "ClashingText")
```

Comment:
Manually renaming this actually fixes the test, so I think this part should be relatively easy to fix. (And I assume a similar fix is needed in gql_collection.go).
```suggestion
		query, err := query.CollectFields(ctx, "NotClashingText")
```